### PR TITLE
many: complete various component cross-checks

### PR DESCRIPTION
**Pull request by andrewphelpsj**
This adds some cross-checks that were missing for components. Testing this was a bit messy unfortunately.

### CODE REVIEW

File: overlord/assertstate/assertmgr.go [hunk 1]
```diff
@@ -215,14 +215,10 @@ func doValidateComponent(t *state.Task, _ *tomb.Tomb) error {
 	}
 
 	db := DB(st)
-	err = snapasserts.CrossCheckResource(compsup.ComponentName(), sha3_384, expectedProv, compSize, compsup.CompSideInfo, snapsup.SideInfo, modelAs, db)
+	resRev, err := snapasserts.CrossCheckResource(compsup.ComponentName(), sha3_384, expectedProv, compSize, compsup.CompSideInfo, snapsup.SideInfo, modelAs, db)
 	if err != nil {
 		return err
```
Comment (pedronis) [inline overlord/assertstate/assertmgr.go:220]: this is already the case but we don't seem to have a test that hits here

**Review by pedronis — COMMENTED** [review]
thanks, test question

File: overlord/assertstate/assertmgr.go (hunk 1 above)
Comment (andrewphelpsj) [reply overlord/assertstate/assertmgr.go:220]: Added a case in 21b440ce49c1f1decf14152ae35338758b4e1a68.

**Review by pedronis — APPROVED** [review]
thanks

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks

**Pull request merged**